tools/pre-commit/verify_set_x.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+
+# Verify no set -x is used in this repository
+
+files="$(git grep -rn --color=always 'set -.*x' -- ':*.sh' ':*/Dockerfile' ':*.yaml' ':*.yml' ":(exclude)$0" ':(exclude).pre-commit-config.yaml')"
+
+if [ -n "$files" ]; then
+    echo "$files" >& 2
+    echo >& 2
+    echo 'error: No script should include "set -x"' >& 2
```

Comment:
Could we give the reason why we should not use `set -x`. Having informative errors messages is great developer experience 😄

Reply:
Indeed but we want to remove this option to avoid leaking secrets in CI logs